podman_tutorial.md

```diff
@@ -0,0 +1,153 @@
+# cd !$ (or cd src/github.com/projectatomic/libpod
+# make
+# sudo make install PREFIX=/usr
+# sudo ln -s /usr/libexec/crio /usr/local/libexec/
+```
+
+<!-- (
+## Install podman on Ubuntu
+
+The default Ubuntu cloud image size will not allow for the following exercise to be done without increasing its
+capacity.  Be sure to add at least 5GB to the image. Instructions to do this are outside the scope of this
+tutorial.
+
+```
+# sudo apt-get update
+# sudo apt-get install libdevmapper-dev libglib2.0-dev libgpgme11-dev golang libseccomp-dev \
+                        go-md2man libprotobuf-dev libprotobuf-c0-dev libseccomp-dev
+# mkdir -p ~/src/github.com/projectatomic/
+# cd ~/src/github.com/projectatomic/
+# git clone https://github.com/projectatomic/libpod/
+#
+# cd libpod
+# make
+# sudo make install PREFIX=/usr
+# sudo mkdir -p /etc/containers
+# sudo bash -c 'cat <<EOF > /etc/containers/registries.conf
+registries.search
+registries = ['docker.io', 'registry.fedoraproject.org']
+EOF
+'
+# sudo bash -c 'cat <<EOF > /etc/containers/policy.json
+ {
+     "default": [
+         {
+             "type": "insecureAcceptAnything"
+         }
+     ],
+     "transports":
+         {
+             "docker-daemon":
+                 {
+                     "": [{"type":"insecureAcceptAnything"}]
+                 }
+         }
+ }
+ EOF
+'
+# git clone https://github.com/containernetworking/plugins.git ~/src/github.com/containernetworking/plugins
+# cd ~/src/github.com/containernetworking/plugins
+# ./build
+# sudo mkdir -p /usr/libexec/cni
+# sudo cp bin/* /usr/libexec/cni
+# git clone https://github.com/opencontainers/runc.git ~/src/github.com/opencontainers/runc
+# GOPATH=~/ make static BUILDTAGS="seccomp selinux"
+# sudo cp runc /usr/bin/runc
+# sudo mkdir -p /usr/local/libexec
+# sudo ln -s /usr/libexec/crio /usr/local/libexec/
+```
+
+) -->
+## Familiarizing yourself with podman
+
+### Running a sample container
+This sample container will run a very basic httpd server that serves only its index
+page.
+```
+# sudo podman run -dt -e HTTPD_VAR_RUN=/var/run/httpd -e HTTPD_MAIN_CONF_D_PATH=/etc/httpd/conf.d \
+                    -e HTTPD_MAIN_CONF_PATH=/etc/httpd/conf \
+                    -e HTTPD_CONTAINER_SCRIPTS_PATH=/usr/share/container-scripts/httpd/ \
+                    registry.fedoraproject.org/f26/httpd /usr/bin/run-httpd
+```
+Note the Container ID returned by that command.
+
+### Listing running containers
+Once you have created or run a container, you can list the known containers with the *ps* command.
+```
+# sudo podman ps
+```
+
+Note: If you add *-a* to the *ps* command, it podman will show all containers.
+
+### Executing a command in a running container
+You can use the *exec* subcommand to execute a command in a running container.  In this case, we
+need to install iproute in the container so we can obtain its IP address.
+```
+# sudo podman exec -t <container_id> dnf install iproute
```

Comment:
This failed for me.  
```
# sudo podman exec -t 46749a3c05d308a269649d1a014137eb53aa110226ba17e7443dc8cab12ae3ce dnf install iproute
Error: Failed to synchronize cache for repo 'updates'
```
Looks to be a dnf error as 'cat /etc/hosts' worked with exec.

Reply:
what failure did you get ?

Reply:
What you see in my prior reply is what I got, error synchronizing the updates repo.

Reply:
It kind of looks like a networking error.  I have seen some strange behaviour where commands like this don't work on their first attempt.  Control-C and repeat seems to work.  Maybe we have a real bug there?

Reply:
And now that I added the urfave/cli option to not reorder args, we can add a -y to the dnf command.

Reply:
FWIW, I set up a new VM and reinstalled podman and I'm still running into this error.

Reply:
It's definitely some kind of network error.   Ctrl-c and retry was no help.  I hopped into the container with /bin/bash and it simply doesn't know how to resolve download.fedora.project. 

What I did to make it work is:

```
# mnt=$(podman mount {containerid})
# dnf -y install --installroot $mnt --release 26 iproute
```

Which might be a nice to add as it shows the flexibility of Podman.

Reply:
@TomSweeneyRedHat  ill hit you up to collaboratively debug.